src/plonkish/compiler/cell_manager.rs

```diff
@@ -299,6 +302,61 @@ impl CellManager for MaxWidthCellManager {
             forward_signal_row
         } as u32;
 
+
+
+        let mut fixed_signal_column: usize = 0;
+        let mut fixed_signal_row: usize = 0;
+        let mut fixed_columns: Vec<Column> = Vec::new();
+
+
+        for fixed_signal in unit.fixed_signals.iter() {
+            let column = if placement.fixed.len() <= fixed_signal_column {
+                let column = if let Some(annotation) = unit.annotations.get(&fixed_signal.uuid())
+                {
+                    Column::fixed(format!("mwcm fixed signal {}", annotation))
+                } else {
+                    Column::fixed("mwcm fixed signal")
+                };
+
+                fixed_columns.push(column.clone());
+                column
+            } else {
+                fixed_columns[fixed_signal_column].clone()
+            };
+
+            placement.fixed.insert(
+                *fixed_signal,
+                SignalPlacement {
+                    column,
+                    rotation: fixed_signal_row as i32,
+                },
+            );
+
+            fixed_signal_column += 1;
+            if fixed_signal_column >= self.max_width_fixed {
+                fixed_signal_column = 0;
+                fixed_signal_row += 1;
+            }
+        }
+
+        placement.base_height = if fixed_signal_row > forward_signal_row {
```

Comment:
@atsupontio you can assign to a new variable max of fixed_signal_row, forward_signal_row and use the result in one expression

Reply:
@leolara Thank you for your comment!
Is it possible to represent it concisely with the following code?

```
let mut max_row = std::cmp::max(fixed_signal_row, forward_signal_row);
placement.base_height = if (fixed_signal_column != 0 && forward_signal_column != 0) {
    max_row as u32 + 1
} else {
    max_row as u32
};
placement.base_height = max_row;
```

Reply:
In that code, you are setting max_row as mut, but you are not mutating it, also you are setting placement.base_height twice. Also, why are you checking both things for zero? If both are zero, max_row will be zero.

Reply:
@leolara Since we don't know whether `fixed_signal_row` or `forward_signal_row` will be assigned to `max_row`, we need to check both `xx_signal_column`, don't we?
The `xx_signal_column` could be 0 even if the `xx_signal_row` is not 0.